docs/vcl-configuration.md

```diff
@@ -121,7 +121,13 @@ backend nginx-backend-6f4c6cbc6c-ckqmv {
 }
 ```
 
-The `.Backends` array includes all backend nodes, no matter if the instance is ready or not. For that reason, it is strongly recommended to set [health checks](https://varnish-cache.org/docs/3.0/tutorial/advanced_backend_servers.html#health-checks) in your VCL configuration. Otherwise, Varnish could send traffic to not yet ready backends.
+The `.Backends` array can include both all backend nodes (default behavior) and only ready nodes. It depends on `.spec.backends.onlyReady` field.
+It depends on which readiness checks you want to respect:
+* Kubernetes readiness checks - field set to `true`. In this case the `.Backends` template var in VCL will include only ready backend pods. 
+Keep in mind that every backend state change will trigger a Varnish VCL config reload.
+* Varnish readiness probes - field set to `false` or omitted. This way the `.Backends` array will include all scheduled pods, so it is strongly recommended to set [health checks](https://varnish-cache.org/docs/3.0/tutorial/advanced_backend_servers.html#health-checks) in your VCL configuration. Otherwise, Varnish could send traffic to not yet ready backends. 
+
+You can also combine approaches and use both Kubernetes and Varnish health probes.
```

Comment:
I think what most people would want out of this config is to have the backends not end up in the backends list until they're ready to serve traffic (not just up), but then they probably still want varnish health checks to route traffic based on a different health check. If the health checks are the same w/the same thresholds, I don't see how it's different aside from having to reload VCL. One could probably assume users are going to enabled both and use the same checks for both. Should we warn against this type of config and maybe add a validation for it? EDIT: the validation would probably be a pain bc of all the parsing required.

Reply:
Not sure if having 2 health checks are useful. I thought I'd mention that possibility, but that's up to the user to use it. I don't think a validation is necessary (and even if necessary it'd be hard to detect those cases reliably).
Maybe one would want to have backup probes in Varnish with longer threshold and rely most of the time on Kubernetes checks.